### ISSUE TITLE
Purpose of Control.Scanl

### ISSUE THREAD

**Issue by YPares** [Gabriel439/Haskell-Foldl-Library#128]
Hi! The intent of this module isn't clear given that `Control.Foldl` already provides `scan`-like functions.
Does it provide extra performance thanks to a more specialized implementation? Could there be an entry about that in this module's documentation?

**Comment by Gabriel439** [comment]
Not everything that can be done with the `Fold` type can be done with the `Scan` type and, vice versa, not everything that can be done with the `Scan` type can be done with the `Fold` type.  In other words, neither type supersedes the other type.

The simplest difference between the two is that:

*   The `Fold` type can produce a result without any input at all

    In other words, `Fold` is an instance of the `Comonad` typeclass

    Conversely, a `Scan` cannot produce a result until it has been fed at least one input

*   `Scan`s can be chained end-to-end

    In other words, `Scan` is an instance of the `Category` typeclass

    Conversely, `Fold`s cannot be chained end-to-end

If that clarifies the difference for you then I can add that explanation to the documentation

**Comment by YPares** [comment]
Yes, thank you, that makes it clearer. I think yes putting it in the doc is a good idea.

**Comment by YPares** [comment]
Also: shouldn't it be always possible to turn a `Scan` into a `Fold` too? I can't find that function.
(It should be possible by setting the fold internal state to `(b,x)` and `extract` to `fst`)

**Comment by Gabriel439** [comment]
@YPares: No, you cannot convert a `Scan` into a `Fold` because a `Scan` cannot provide a function of type `x -> b` that a `Fold` requires

**Comment by YPares** [comment]
But cannot this function be just `id` here? Since in that case `b ~ x`, right?
